### PR TITLE
fix(sampler): change xtc_threshold default from 0.0 to 0.1

### LINKED ISSUE #1036: XTC threshold=0.0 default causes destructive sampling
## Summary

When a user sets `xtc_probability` without explicitly setting `xtc_threshold`, the default `xtc_threshold=0.0` causes `apply_xtc()` to mask all tokens except the least probable one. This is almost certainly not the intended behavior.

## Root cause

In `sample_utils.py:17`, `xtc_threshold` defaults to `0.0`:

```python
def make_sampler(
    ...
    xtc_probability: float = 0.0,
    xtc_threshold: float = 0.0,   # <-- here
    ...
```

The activation gate at `sample_utils.py:55` only checks `xtc_probability`:

```python
if xtc_probability > 0.0:
    sampling_methods.append(
        lambda x: apply_xtc(x, xtc_probability, xtc_threshold, xtc_special_tokens)
    )
```

Inside `apply_xtc()` at `sample_utils.py:280-281`:

```python
probs = mx.softmax(logits, -1)
mask = probs > mx.where(probs > xtc_threshold, probs, mx.inf).min()
```

When `xtc_threshold=0.0`, all softmax probabilities are > 0.0, so `mx.where(probs > 0.0, probs, mx.inf)` returns probs unchanged. `.min()` then yields the smallest probability in the distribution. The mask selects every token with probability above that minimum, meaning all tokens except the single least probable one get masked to `-inf`.

With `xtc_probability=0.5`, this means 50% of token samples will be forced to select from only the least probable token(s).

## Steps to reproduce

```python
from mlx_lm.sample_utils import make_sampler

# User sets probability but relies on default threshold
sampler = make_sampler(temp=0.7, xtc_probability=0.5)
# Result: XTC is active with threshold=0.0, which is maximally aggressive
```

## Same default in server.py

`server.py:1233` also defaults to `0.0`:

```python
self.xtc_threshold = self.body.get("xtc_threshold", 0.0)
```

So API callers that send `{"xtc_probability": 0.5}` without `xtc_threshold` hit the same issue.

## Suggested fix

Default `xtc_threshold` to `0.1` (consistent with the XTC author's recommended range) in both `make_sampler()` and the server request parser. This way, setting only `xtc_probability` produces reasonable behavior out of the box.

Alternatively, if `xtc_threshold=0.0` is meant to disable XTC independently of `xtc_probability`, the activation gate at `sample_utils.py:55` should also check `xtc_threshold > 0.0`.

### PR DESCRIPTION
**Pull request by 0xSoftBoi**
## Problem
With `xtc_threshold=0.0` (old default) and `xtc_probability>0`, `apply_xtc()` masks every token above probability 0.0 — i.e. all tokens except the single least-probable one. This is the opposite of the intended behaviour and produces degenerate output whenever a user sets `xtc_probability` without an explicit threshold.

## Fix
Change the default to `0.1`, matching the threshold recommended in the original XTC paper and used in all documented examples.

Fixes #1036

### CODE REVIEW

**Comment by 0xSoftBoi** [comment]
Superseded by #1372 which also fixes the CLI default in generate.py and the API body default in server.py.
